src/plots/cartesian/axes.js

```diff
@@ -1683,12 +1682,16 @@ axes.doTicks = function(gd, axid, skipTitle) {
                 counterAxis._length + 10 +
                     fontSize*(offsetBase + (ax.showticklabels ? 1.5 : 0.5)));
 
+            if(ax.rangeslider && ax.rangeslider.visible && ax._boundingBox) {
+                y += gs.h * ax.rangeslider.thickness + ax._boundingBox.height;
```

Comment:
@mdtusz this moved over here from `Titles.draw` when I merged master in with rangesliders. I noticed you recalculated the inner plot height, but it's better (and shorter) to use `fullLayout._size` which has this cached, AFTER anything that may potentially have increased the margin (like an external colorbar or legend)